src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -596,6 +589,23 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         .tags(TagUtils.sanitizeTags(snapshotRequestModel.getTags()));
   }
 
+  private void getTablesByQueryAsset(
+      String datasetName, String assetName, SnapshotSource snapshotSource, Snapshot snapshot) {
+    Dataset queryDataset = datasetService.retrieveByName(datasetName);
+    AssetSpecification queryAssetSpecification =
+        queryDataset
+            .getAssetSpecificationByName(assetName)
+            .orElseThrow(
+                () ->
+                    new AssetNotFoundException(
+                        "This dataset does not have an asset specification with name: "
+                            + assetName));
+    snapshotSource.assetSpecification(queryAssetSpecification);
+    // TODO this is wrong? why don't we just pass the assetSpecification?
+    conjureSnapshotTablesFromAsset(
+        snapshotSource.getAssetSpecification(), snapshot, snapshotSource);
```

Comment:
To answer this `TODO`, I would remove the `snapshotSource.getAssetSpecification()` parameter and have `conjureSnapshotTablesFromAsset()` call `snapshotSource.getAssetSpecification()` internally.

The reason why it's set on snapshot source at all is because we're building the `Snapshot` object that will be written to the database once this step is complete.